src/utilities/runCompletion.ts

```diff
@@ -24,10 +24,22 @@ import { sleep } from "./utils";
 import acceptDisclaimer from "../commands/acceptDisclaimer";
 import { getServiceApi } from "../services/common";
 import { clearPromptFeedbackCodeLens } from "../codelens/FeedbackCodelensProvider";
+import { StreamingPipeline } from "./streamingPipeline";
 
 let cancelCompletion: AbortController | null = null;
 let promptId: string | undefined = undefined;
 
+/**
+ * Cancel the current completion request
+ * Used when user accepts a completion early or manually cancels
+ */
+export function cancelCurrentCompletion(): void {
+  if (cancelCompletion) {
+    cancelCompletion.abort();
+    cancelCompletion = null;
+  }
+}
+
```

Comment:
when cancelling the completion early i see the stream stop but the spinner never stops until i run another completion and let that finish normally